coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                    cls._get_process_output_metadata(),
+                    cls._get_generate_config_metadata(),
+                    cls._get_create_arguments_metadata())
+                merged_metadata.desc = inspect.getdoc(cls)
+                return merged_metadata
+
+            @classmethod
+            def _execute_command(cls, args, stdin=None):
+                """
+                Executes the underlying tool with the given arguments.
+
+                :param args:
+                    The argument sequence to pass to the executable.
+                :param stdin:
+                    Input to send to the opened process as stdin.
+                :return:
+                    A tuple with ``(stdout, stderr)``.
+                """
+                return run_shell_command(
+                    (cls.get_executable(),) + tuple(args),
+                    stdin=stdin)
+
+            def _convert_output_regex_match_to_result(self,
+                                                      match,
+                                                      filename,
+                                                      severity_map):
+                """
+                Converts the matched named-groups of ``output_regex`` to an
+                actual ``Result``.
+
+                :param match:
+                    The regex match object.
+                :param filename:
+                    The name of the file this match belongs to.
+                :param severity_map:
+                    The dict to use to map the severity-match to an actual
+                    ``RESULT_SEVERITY``.
+                """
+                # Pre process the groups
+                groups = match.groupdict()
+
+                try:
+                    groups["severity"] = severity_map[
+                        groups["severity"].lower()]
+                except KeyError:
+                    self.warn(
+                        "No correspondence for " + repr(groups["severity"]) +
+                        " found in given severity map. Assuming "
+                        "`RESULT_SEVERITY.NORMAL`.")
+                    groups["severity"] = RESULT_SEVERITY.NORMAL
+
+                for variable in ("line", "column", "end_line", "end_column"):
+                    groups[variable] = (None
+                                        if groups.get(variable, "") == "" else
+                                        int(groups[variable]))
+
+                if "origin" in groups:
+                    groups["origin"] = "{} ({})".format(
+                        str(klass.__name__),
+                        str(groups["origin"]))
+
+                # Construct the result.
+                return Result.from_values(
+                    origin=groups.get("origin", self),
+                    message=groups.get("message", ""),
+                    file=filename,
+                    severity=int(groups.get("severity",
```

Comment:
I think L446 ensures that the severity will definitely have a value and this default value of NORMAL is not needed